datasets/medical_dialog/medical_dialog.py

```diff
@@ -0,0 +1,280 @@
+                            last_part = "dialogue"
+                            if data_lang == "zh":
+                                last_user = "病人"
+
+                            if data_lang == "en":
+                                last_user = "Patient"
+
+                            while True:
+                                line = f_in.readline()
+                                if (not line) or (line in ["\n", "\n\r"]):
+                                    conv_flag = False
+                                    last_user = ""
+                                    last_list.append(copy.deepcopy(last_conv))
+                                    # To ensure close of conversation, only even number of sentences
+                                    # are extracted
+                                    last_turn = len(last_list)
+                                    if int(last_turn / 2) > 0:
+                                        temp = int(last_turn / 2)
+                                        id_ += 1
+                                        last_dialog["file_name"] = filepath
+                                        last_dialog["dialogue_id"] = dialogue_id
+                                        last_dialog["dialogue_url"] = dialogue_url
+                                        last_dialog["dialogue_turns"] = last_list[: temp * 2]
+                                        yield id_, last_dialog
+                                    break
+
+                                if data_lang == "zh":
+                                    if line[:3] == "病人：" or line[:3] == "医生：":  # Hardcode alert!
+                                        user = line[:2]  # Hardcode alert!
+                                        line = f_in.readline()
+                                        conv_flag = True
+
+                                # The elif block is to ensure that multi-line sentences are captured.
+                                # This has been observed only in english.
+                                if data_lang == "en":
+                                    if line.strip() == "Patient:" or line.strip() == "Doctor:":  # Hardcode alert!
+                                        user = line.replace(":", "").rstrip()
+                                        line = f_in.readline()
+                                        conv_flag = True
+                                    elif line[:2] != "id":  # Hardcode alert!
+                                        conv_flag = True
+
+                                # Continues till the next ID is parsed
+                                if conv_flag:
+                                    sen = line.rstrip()
+                                    if sen == "":
+                                        continue
+
+                                    if sen[-1] not in ".。？，,?!！~～":
+                                        sen += "."
```

Comment:
I don't think we should re-add the dots here.
The processing of the dataset is left to the user, here we're just supposed to parse the data files